controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -167,8 +165,44 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 		}
 
 		// TODO: Phases do not have checks at the moment, in order to make them more predictable and less error-prone, at the expense of increased computation cost.
-		// Creation phase.
-		r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimCreating
+		// Validation phase.
+		r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimValidating
+
+		// If a StorageClass already exists:
+		// 	StorageClassClaim passes validation and is promoted to the configuring phase if:
+		//  * the StorageClassClaim has the same type as the StorageClass.
+		// 	* the StorageClassClaim has no encryption method specified when the type is filesystem.
+		// 	* the StorageClassClaim has a blockpool type and:
+		// 		 * the StorageClassClaim has an encryption method specified.
+		// 	  * the StorageClassClaim has the same encryption method as the StorageClass.
+		// 	StorageClassClaim fails validation and falls back to a failed phase indefinitely (no reconciliation happens).
+		existing := &storagev1.StorageClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: r.storageClassClaim.Name,
+			},
+		}
+		if err = r.get(existing); errors.IsAlreadyExists(err) {
```

Comment:
This error check is wrong any error other than not found needs to return, if err is nil means we found storageclass and then do the comparison.

Reply:
@Madhu-1 Changed to `if err = r.get(existing); !errors.IsNotFound(err) {`. This will catch it if it's `nil`. Is this okay?